stories/core/FileUpload.stories.tsx

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { SvgUpload } from '@itwin/itwinui-icons-react';
+import { action } from '@storybook/addon-actions';
+import { useState } from '@storybook/client-api';
+import { Story, Meta } from '@storybook/react';
+import React from 'react';
+import { FileUpload, FileUploadProps, Textarea } from '../../src/core';
+
+export default {
+  component: FileUpload,
+  argTypes: {
+    className: { control: { disable: true } },
+    style: { control: { disable: true } },
+    onFileDropped: { control: { disable: true } },
+    children: { control: { disable: true } },
+  },
+  title: 'Core/FileUpload',
+} as Meta<FileUploadProps>;
+
+export const Basic: Story<FileUploadProps> = (args) => {
+  const [files, setFiles] = useState<Array<File>>([]);
+
+  return (
+    <FileUpload
+      content={'Drop file to upload'}
+      {...args}
+      onFileDropped={(files) => {
+        setFiles(Array.from(files));
+        action(`${files.length} files uploaded`)();
+      }}
+    >
+      <Textarea
+        placeholder={'Drag a file here'}
+        rows={1}
+        defaultValue={files.map((f) => f.name)}
+      />
+    </FileUpload>
+  );
+};
+
+Basic.args = {
+  content: 'Drop file to upload',
+};
+
+export const WithoutChildren: Story<FileUploadProps> = (args) => {
+  const [files, setFiles] = useState<Array<File>>([]);
+
+  return (
+    <FileUpload
+      content={
+        <>
```

Comment:
It would be nice to provide this from our side but I am not sure how...

Reply:
So users will need to copy paste this one themselves? I would agree we need to render this content ourselves...

Reply:
But it's so basic. If we provide something, we should go with a better template and it also needs to be flexible

Reply:
If there are classes user need to add, it is not basic anymore...

Reply:
It is pretty basic. I think we decided that if we are going to provide something, we should go all the way. i.e. include upload tracker, final state, etc

For now, the main purpose of this component is to wrap user content/children

Reply:
I would agree, if this was our LabeledInput..Now its just plain html and most important thing, it needs our class to be applied. It is just a blocker in here, IMO.
We removed simple things (applying iui-icon class requirement), we cannot expect this to work and I imagine this workflow might be used more often than with wrapper element..

Reply:
Added `FileUploadTemplate` based on https://github.com/iTwin/iTwinUI/pull/54